frontend/src/components/settings/SettingInput.tsx

```diff
@@ -115,19 +123,33 @@ const SettingInput: React.FC<RenderSettingInputProps> = ({
         />
       );
     case "select": {
-      if (setting.label === "fallback_block") {
+      if (setting.config?.["entity"] === "Block") {
         const { onChange, ...rest } = field;
+        const config = setting.config as any;
 
         return (
           <AutoCompleteEntitySelect<IBlock, "name", false>
             searchFields={["name"]}
-            entity={EntityType.BLOCK}
-            format={Format.BASIC}
-            labelKey="name"
-            label={t("label.fallback_message")}
-            helperText={t("help.fallback_message")}
-            multiple={false}
-            onChange={(_e, selected, ..._) => onChange(selected?.id || "")}
+            format={Format.FULL}
+            label={label}
+            groupBy={({ category }) =>
+              getCategoryFromCache(category)?.label ?? t("label.other")
+            }
+            onChange={(_e, selected) => onChange(selected?.id)}
+            renderGroup={({ key, group, children }) => (
+              <li key={key}>
+                <Typography
+                  component="h4"
+                  p={2}
+                  fontWeight={700}
+                  color="primary"
+                >
+                  {group}
+                </Typography>
+                <Box>{children}</Box>
+              </li>
+            )}
+            {...config}
```

Comment:
@yassinedorbozgithub Are you sure the "config" attributes are valid props to the component ?